src/ansible_navigator/utils/serialize.py

```diff
@@ -2,11 +2,27 @@
 
 import json
 
+from dataclasses import asdict
+from dataclasses import is_dataclass
 from typing import IO
 from typing import Any
 from typing import NamedTuple
 
 
+class EnhancedJSONEncoder(json.JSONEncoder):
+    """An enhanced json encoder accounting for dataclasses."""
```

Comment:
How about being clearer here? Abstract statements like "accounting for XXX" don't answer the "how? what?" question.
```suggestion
    """An enhanced JSON encoder rendering dataclasses as :class:`dicts <dict>`."""
```

Reply:
FYI, this will be in a separate PR, #1014